query-translator/src/types.rs

```diff
@@ -25,37 +27,57 @@ use mentat_sql::{
     QueryBuilder,
     QueryFragment,
     SQLiteQueryBuilder,
+    SQLQuery,
 };
 
 //---------------------------------------------------------
 // A Mentat-focused representation of a SQL query.
 
-enum ColumnOrExpression {
+pub enum ColumnOrExpression {
     Column(QualifiedAlias),
-    Integer(i64),             // Because it's so common.
+    Integer(i64),            // Because it's so common.
+    Value(TypedValue),       // TODO: replace with a SQLite value.
```

Comment:
A note that the intention is to have this independent of the Mentat data types (if this is true, which I think it is) would be nice.